openedx/core/djangoapps/lang_pref/middleware.py

```diff
@@ -44,6 +44,10 @@ def process_request(self, request):
                 accept_header = cookie_lang
             request.META[LANGUAGE_HEADER] = accept_header
 
+            # Allow the new cookie setting to update the language in the user's session
+            if LANGUAGE_SESSION_KEY in request.session and request.session[LANGUAGE_SESSION_KEY] != cookie_lang:
+                del request.session[LANGUAGE_SESSION_KEY]
```

Comment:
I see, could there ever be a case where the cookie is invalid and we want to fallback to whatever had previously been in the session?